src/util/events.rs

```diff
@@ -100,6 +100,13 @@ pub enum Event {
 	BroadcastChannelUpdate {
 		msg: msgs::ChannelUpdate,
 	},
+	/// Used to tell a peer that something is incorrect
+	SendErrorMessage {
```

Comment:
Ok change as recommanded, but add two flags, one for disconnection and another one for a "hard" disconnect, when you want you to disconnect without spoiling ressources on an error message , in case of banned peers or malicious ones.